architect/systems/spectrometers.py

```diff
@@ -77,40 +78,38 @@ def get_ratio_cropped_light_through_slit(self):
 
     def get_signal_to_noise(self, radiance: LUT, wavelength):
         """Get the signal to noise ratio of the system.
-
         Ref: https://www.notion.so/utat-ss/Signal-to-Noise-6a3a5b8b744d41ada40410d5251cc8ac
-
         """
 
-        snr = signal / self.get_noise()
+        snr = (
+            self.get_signal(radiance=radiance, wavelength=wavelength) / self.get_noise(wavelength, radiance)
```

Comment:
Let's add the parameter names for the sake of explicitness
```suggestion
            self.get_signal(radiance=radiance, wavelength=wavelength) / self.get_noise(wavelength=wavelength, radiance=radiance)
```